dev/bots/prepare_package.dart

```diff
@@ -255,6 +264,8 @@ class ArchiveCreator {
   late File _outputFile;
   final Map<String, String> _version = <String, String>{};
   late String _flutter;
+  late String _dart;
```

Comment:
This isn't really related to the arch change, but a better pattern than the `late` ivars would probably be a factory constructor:
```dart
  factory ArchiveCreator.archive(
    Directory tempDir,
    Directory outputDir,
    String revision,
    Branch branch, {
    bool strict = true,
    ProcessManager? processManager,
    bool subprocessOutput = true,
    Platform platform = const LocalPlatform(),
    HttpReader? httpReader,
  })  {
    final Directory flutterRoot = Directory(path.join(tempDir.path, 'flutter'));
    final ProcessRunner processRunner = ProcessRunner(
      processManager: processManager,
      subprocessOutput: subprocessOutput,
      platform: platform,
    );
    processRunner.environment['PUB_CACHE'] = path.join(flutterRoot.absolute.path, '.pub-cache');

    final String flutterExecutable = path.join(
      flutterRoot.absolute.path,
      'bin',
      'flutter',
    );
    final String dartExecutable = path.join(
      flutterRoot.absolute.path,
      'bin',
      'cache',
      'dart-sdk',
      'bin',
      'dart',
    );
    return ArchiveCreator._(
      tempDir: tempDir,
      platform: platform,
      flutterRoot: flutterRoot,
      outputDir: outputDir,
      revision: revision,
      branch: branch,
      strict: strict,
      processRunner: processRunner,
      httpReader: httpReader ?? http.readBytes,
      flutterExecutable: flutterExecutable,
      dartExecutable: dartExecutable,
    );
  }
```
```dart
  ArchiveCreator._({
    required this.tempDir,
    required this.platform,
    required this.flutterRoot,
    required this.outputDir,
    required this.revision,
    required this.branch,
    required this.strict,
    required ProcessRunner processRunner,
    required this.httpReader,
    required String flutterExecutable,
    required String dartExecutable,
  })  : assert(revision.length == 40),
        _processRunner = processRunner,
        _flutter = flutterExecutable,
        _dart = dartExecutable;
```

Reply:
Good suggestion. Done.